### PR TITLE
Feat/update code after review

### PR DESCRIPTION
**Pull request by tranchinh123**

### CODE REVIEW

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 1]
```diff
@@ -1,27 +1,19 @@
...
   return (
     <div className={styles.bookList}>
       {books.map((book) => (
-        <BookCard
-          key={book.id}
-          book={book}
-          onHandleClick={() => onHandleClick(book)}
-        />
+        <Link key={book.id} to={`/detail/${book.id}`}>
+          <BookCard key={book.id} book={book} />
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx:15]: remove key in Book

File: bookshelter-app/src/components/ButtonBack/index.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import styles from './index.module.css';
+import LeftArrow from '../Icons/LeftArrow';
+
+interface ButtonBackProps {
+  handleBackClick: () => void;
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/ButtonBack/index.tsx:5]: use on prefix

File: bookshelter-app/src/components/ButtonBack/index.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+const ButtonBack = ({ handleBackClick }: ButtonBackProps) => {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/ButtonBack/index.tsx:8]: BackButton

File: bookshelter-app/src/components/CategoryList/index.tsx [hunk 4]
```diff
@@ -69,13 +68,14 @@ const CategoryList = ({
...
 
       {categories.map((category, index) => (
-        <BookCategory
-          key={category.id}
-          name={category.name}
-          totalBooks={category.totalBooks}
-          color={colors[index]}
-          onClick={() => handleCategoryClick(category.slug)}
-        />
+        <Link key={category.id} to={`/${category.slug}`}>
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/CategoryList/index.tsx]: Wrap Link in BookCategory component

File: bookshelter-app/src/components/BookCardList/index.tsx (hunk 1 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx:15]: if remove key console Warning: Each child in a list should have a unique "key" prop.

File: bookshelter-app/src/components/ButtonBack/index.tsx (hunk 2 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/ButtonBack/index.tsx:5]: Fixed!

File: bookshelter-app/src/components/ButtonBack/index.tsx (hunk 3 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/ButtonBack/index.tsx:8]: Fixed!

File: bookshelter-app/src/components/CategoryList/index.tsx (hunk 4 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/CategoryList/index.tsx]: Fixed!

File: bookshelter-app/src/components/BookCategory/index.tsx [hunk 5]
```diff
@@ -4,16 +4,16 @@ interface BookCategoryProps {
...
+  onClick: () => void;
 }
 
 const BookCategory = ({
   name,
   totalBooks,
   color,
-  onClick,
   categorySlug,
+  onClick,
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCategory/index.tsx:16]: remove it if not use

Comment (tranchinh123) [reply bookshelter-app/src/components/BookCategory/index.tsx:16]: when user click BookCategory in Mobile app and close CategoryList

File: bookshelter-app/src/components/BookCardList/index.tsx (hunk 1 above)
Comment (huypham-agilityio) [reply bookshelter-app/src/components/BookCardList/index.tsx:15]: Link already has the key.

**Pull request merged**